usagov/usagov-2021
# Usagov 603 import directory synonyms

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-603

## Description

"Synonyms" are separate entities from the Federal Directory Records they refer to, so they need to be imported as a separate step. The code changes here are within the usagov_directories custom module and include: 

* modified the agency_import_prep.php script to generate a "synonyms.csv" when it runs
* added a form to which a CSV file of synonyms can be uploaded, which creates the appropriate Federal Directory Synonyms
* added menu entries (in the admin area) for the above form and for some already-existing forms in the usagov_directories module. 

I also updated some of the documentation. 

## To test: 

Follow the instructions at https://github.com/usagov/usagov-2021/blob/USAGOV-603-import-directory-synonyms/web/modules/custom/usagov_directories/docs/Importing_Federal_Agency_Records.md 

If you have _not_ already imported Federal Directory Records, start at **Importing the data to beta (first time).** Processed CSV files are available for download from https://drive.google.com/drive/folders/1xw33rd_TGSGA9ExF1QX2EOjnzFWMMEgx -- in the "processed" folder. (Note that this lets you skip downloading the files from mothership and running the agency_import_prep.php script.) 

**If you have already imported Federal Directory Records**, download the "synonyms.csv" file from the "processed" folder (link above), and just follow the instructions under **Importing the synonyms.**

You should see: 

* an encouraging tally of imported synonyms (you will have at least one "skipped," because in the best case, the import of Directory Records will have rejected the one for USA.gov). 
* in admin -> Content, a number of nodes of type "Federal Directory Synonym." Spot check; each should have a title, language, selected, and an Agency Reference that seems plausible. 

(You should also be able to find the English synonyms listed appropriately under /agency-index, but that's a separate bit of work.)  

## Checklist for the Developer



## Checklist for the Peer Reviewers
